src/components/Button/Button.tsx

```diff
@@ -185,7 +185,11 @@ export function Button({
     ) : (
       icon
     );
-    iconMarkup = <span className={styles.Icon}>{iconInner}</span>;
+    iconMarkup = (
+      <span className={classNames(styles.Icon, loading && styles.Hidden)}>
+        {iconInner}
+      </span>
+    );
```

Comment:
I went with hiding the icon to preserve the width of the button, instead of removing it entirely.